lib/ansible/template/native_helpers.py

```diff
@@ -62,6 +66,16 @@ def ansible_native_concat(nodes):
         # TODO send unvaulted data to literal_eval?
         if isinstance(out, AnsibleVaultEncryptedUnicode):
             return out.data
+
+        if isinstance(out, NativeJinjaText):
+            # Sometimes (e.g. ``| string``) we need to mark variables
+            # in a special way so that they remain strings and are not
+            # passed into literal_eval.
+            # See:
+            # https://github.com/ansible/ansible/issues/70831
+            # https://github.com/pallets/jinja/issues/1200
+            # https://github.com/ansible/ansible/issues/70831#issuecomment-664190894
+            return out
```

Comment:
After looking at the added integration tests, I wonder if we should do `return text_type(out)` so that we "unwrap" the var at this point.  I don't think it should matter, but I also don't think we should expose the implementation details of how we achieve this to users.  I think using `text_type` instead of `to_text` would be required.

Reply:
I agree we should not expose the implementation details however it is kept wrapped on purpose so we can do:
```yaml
compose:
  ec2_account_id: 1234567890 | string
keyed_groups:
- key: ec2_account_id
  parent_group: accounts
  prefix: ''
  separator: ''
plugin: constructed
```
If we unwrapped the var in `ansible_native_concat` then `ec2_account_id` in `- key:` would be run through `literal_eval` again which would result in the error from the original issue. So we need to preserve the `NativeJinjaText` information across `template()` calls.

Regardless of (un)wrapping this would still work though:
```yaml
compose:
  ec2_account_id: 1234567890
keyed_groups:
- key: ec2_account_id | string
  parent_group: accounts
  prefix: ''
  separator: ''
plugin: constructed
```

Reply:
Ok, that was the concern I had, is that this wasn't "final" enough in this location.  That it was intermediate.